src/overlays/actors/ovl_Bg_Gnd_Darkmeiro/z_bg_gnd_darkmeiro.c

```diff
@@ -35,25 +35,225 @@ const ActorInit Bg_Gnd_Darkmeiro_InitVars = {
+				thisx->draw = BgGndDarkmeiro_DrawSwitchedPlatform;
+				this->actionFunc = BgGndDarkmeiro_SwitchedPlatform;
+				if (Flags_GetSwitch(globalCtx, ( thisx->params >> 8) & 0x3F) == 0) {
+					func_8003EBF8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId); //Turn off collision if switch is off
+				} else {
+					this->timer1 = 0x40; //Set vanishing timer if switch is on
+					this->actorState |=  2; //Set active state if switch is on
+				}
+			}
+			break;
+		case 2: /* Transparent platform timer.
+			A transparent platform timer with switch flag N reacts to switch flags N+1 and N+2 being set, setting its own switch flag and a timer for 0x130 frames. There are separate timers for N+1 and N+2, and the timer sets flag N as long as either timer is above 0x40. Despite the name, it does not require any transparent platforms to be present to work.
+			*/
+			this->actorState = this->timer1 = this->timer2 = 0; //initialize state and timers to 0
+			this->actionFunc = BgGndDarkmeiro_PlatformTimer;
+			thisx->draw = NULL; 
+			if (Flags_GetSwitch(globalCtx, (( thisx->params >> 8) & 0x3F) + 1) != 0) {
+				this->timer1 = 0x40; //If switch N+1 is already set, turn on timer 1 with 0x40 frames on the clock.
+				this->actorState |= 4;
+			}
+			if (Flags_GetSwitch(globalCtx, (( thisx->params >> 8) & 0x3F) + 2) != 0) {
+				this->timer2 = 0x40; //If switch N+2 is already set, turn on timer 2 with 0x40 frames on the clock.
+				this->actorState |= 8; 
+			}
+			if ((this->timer1 != 0)||(this->timer2 != 0)) {
+				Flags_SetSwitch(globalCtx, ( thisx->params >> 8) & 0x3F); //If the timers are set, set switch N
+			} else {
+				Flags_UnsetSwitch(globalCtx, ( thisx->params >> 8) & 0x3F); //If not, clear it.
+			}
+			break;
+	}
+	if (globalCtx){}; //needed for matching
+}
+
+//#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/BgGndDarkmeiro_Init.s")
+
+void BgGndDarkmeiro_Destroy(Actor *thisx, GlobalContext* globalCtx) {
+	BgGndDarkmeiro* this = THIS;
+	
+    if ((this->dyna.actor.params & 0xFF) == 1) {
+		if (!globalCtx){}; //needed for matching
+        DynaPolyInfo_Free(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+    }
+}
+
+// #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/BgGndDarkmeiro_Destroy.s")
+
+void BgGndDarkmeiro_NoAction(BgGndDarkmeiro *this, GlobalContext *globalCtx) {
+	//This is mode 0's action function. As you can see, it does nothing.
+}
+
+//#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Gnd_Darkmeiro/BgGndDarkmeiro_NoAction.s")
+
+void BgGndDarkmeiro_PlatformTimer(BgGndDarkmeiro *this, GlobalContext *globalCtx) {
+    s16 phi_a1;
+
+    if (Flags_GetSwitch(globalCtx, ((this->dyna.actor.params >> 8) & 0x3F) + 1) != 0) { //If switch N+1 is active...
+        if ((this->actorState & 4) != 0) { // and timer 1 is running...
+            if (this->timer1 > 0) {//if there's any time left...
+                this->timer1 -= 1; //decrement timer 1.
```

Comment:
This is very minor but in C you can use the `--` operator.
```suggestion
                this->timer1--; //decrement timer 1.
```